api/app/clients/sms/telstra.py

```diff
@@ -51,11 +53,31 @@ def send_sms(self, to, content, reference, sender=None):
             notify_url=notify_url
         )
 
+        # Avoid circular imports by importing this file later.
+        from app.models import (
+            NOTIFICATION_SENDING,
+            NOTIFICATION_PERMANENT_FAILURE,
+        )
+
         try:
             resp = telstra_api.send_sms(payload)
             message = resp.messages[0]
             self.logger.info(f"Telstra send SMS request for {reference} succeeded: {message.message_id}")
 
+            return message.message_id, NOTIFICATION_SENDING
+        except telstra.rest.ApiException as e:
+            try:
+                messages = json.loads(e.body)["messages"]
+
+                if e.status == 400 and len(messages) > 0 and messages[0]["code"] == "TO-MSISDN-NOT-VALID" and messages[0]["deliveryStatus"] == "DeliveryImpossible":
```

Comment:
we'll need to cover other error types here when possible

Reply:
yeah this line is a bit meh, but the alternative was to prematurely pull this out into a general function with consts, etc. but seemed overkill for now